### DISCUSSION TITLE
egg 54184 not working (builtin wifi)

### DISCUSSION THREAD

**Discussion by wyliecoyoteuk** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#364] [General]
Hi all, I have 2 eggs, one works perfectly.
The other, model 54184 has a bestway wifi module.
switched on, it plays the tune, displays the IP address and date, then nothing for a while, then displays 25 as the temperature.
The bubbles came on then the website went offline, physical buttons don't work either.
Has anyone else managed to get a wifi pump working?

**Comment by visualapproach** [comment]
Someone was tinkering with a Wi-Fi pump but I don't remember if it worked or not. But you should use pcb_v2B (you can use any physical pcb but use the pinout according to this)
Also best chances with the transistor level converter (red) but both may work.

**Reply by wyliecoyoteuk** [reply]
I have V1 PCBs and the red LLC.

For your compatibility list, The working pump is a Hawaii airjet model 54154
using model pre 2021

The no working 54184 has 6 wires

**Reply by wyliecoyoteuk** [reply]
BTW, sent you a coffee

**Reply by visualapproach** [reply]
Lovely! Thank you for your support! 🙏 

Did you try with different pinout on the 54184? Define PCB_V2B in the code, and connect according to the comments in `model.h`:
CIO_DATA (wire #3) : D1 (port H1A)
CIO_CLK (wire #4)  : D2 (port H1A)
CIO_CS (wire #5)   : D5 (port H2A)

DSP_DATA (wire #3) : D6 (port H2A)
DSP_CLK (wire #4)  : D4 (port H2A)
DSP_CS (wire #5)   : D3 (port H1A)
```
#define PCB_V2B  
```

**Reply by wyliecoyoteuk** [reply]
We have guests for the next few days, so I will try that next week.

**Reply by wyliecoyoteuk** [reply]
Tried the new pinouts, start up tune, no display, web interface shows no temperature, buttons work intermittently ( e.g. pump switches on, goes off again, next try switches on and stays on, won't switch off. bubbles come on and go off straight after.

**Reply by wyliecoyoteuk** [reply]
TBH, the result with the original PCB was better, at least it was obviously passing information to the display.

**Reply by visualapproach** [reply]
Does it pass the edited: hwtestinfo.html? (Dev branch, don't forget to upload the data/*.* files)

**Reply by visualapproach** [reply]
you can do it on the v2 too ofc

**Reply by wyliecoyoteuk** [reply]
I'll have to look at that.

**Reply by wyliecoyoteuk** [reply]
DSP input pin 2 0 errors of 1000
DSP input pin 3 0 errors of 1000
DSP input pin 4 0 errors of 1000
CIO input pin 2 0 errors of 1000
CIO input pin 3 0 errors of 1000
CIO input pin 4 0 errors of 1000
That is the result on both pinouts (I have 2 v1 pcbs wired as v1 and a v1 pcb wired as v2b)

**Reply by visualapproach** [reply]
So your ESP board is okay. I don't know what's wrong to be honest. Seems the signals are corrupted. Anyone else having an idea? Maybe they have 3.3V signals now as an adaptation to the ESP32 they put in? In that case you wouldn't want the LLC. (It's not strictly needed for 5V either)

**Reply by wyliecoyoteuk** [reply]
The 3.3v idea makes sense.
I'll check it out with a meter later

**Reply by visualapproach** [reply]
Yes, the power may or may not be 5V so check the other wires also.

**Reply by wyliecoyoteuk** [reply]
Nope, 4.9V on all signal wires.
details:
V1PCB V1 pinouts:
pump display:The tune plays, display shows IP address and date, then nothing. filter button has a dim light showing. No buttons work
Web interface: shows temp correctly. the filter and air buttons work, but after a 10-20 second delay, and they don't show as working on the web page (i.e. The button goes green then grey again, then about 10 seconds later the filter comes on.)
Incidentally it seems to disable the onboard esp32 unit.

**Reply by visualapproach** [reply]
I can only think of three ways the 8266 could knock out the 32;
- power demand is more than the pump can deliver on the 5V rail. Would be testable with separate power. If you try, disconnect pin 1 (5V) from the CIO.
- Wi-Fi interference.
- CIO confused about weird button codes from the 8266 and so it shuts down the Wi-Fi.
- 8266 overloads the CIO. Not possible since the CIO is master.

You can check in the browser console or mqtt what buttons the 8266 thinks you are pressing on the display. I don't think that's the problem but worth checking.

**Reply by wyliecoyoteuk** [reply]
It is possible that I reset it by accident while trying buttons, it works without the module connected.
Some pics of the CIO and the wifi connection, would seem to suggest ESP32 unit is running on 5v.
![image](https://user-images.githubusercontent.com/79256941/187076384-9293088d-8b42-4d98-9c0a-92f2233c8cd9.png)
![image](https://user-images.githubusercontent.com/79256941/187076398-8000fb2d-fe7c-427d-b348-eb369da7ebc3.png)

**Reply by wyliecoyoteuk** [reply]
TBH, I wish I could disable the built-in wifi, it is almost impossible to connect to, and very limited.

**Reply by wyliecoyoteuk** [reply]
Update, After a struggle requiring a very old version of android, I got the Bestway app connecting again. https://wyliecoyoteuk.wordpress.com/2022/05/09/why-does-android-block-some-iot-wifi-devices/
With the v1 board plugged in, the buttons on the display don't work, however they light up correctly when you switch on the pump or filter using either the Bestway app or the Web interface. 
However response to either is erratic, sometimes they will switch on and then off again after a few seconds, sometimes not. The Bestway app sometimes seems to get confused and goes to the power on screen. 
So I have the Bestway option as a backup in case the other pump fails. Unfortunately the pump I am using has a 2 degree hysteresis and lacks the freezeshield function, But I can live with it.

**Reply by visualapproach** [reply]
> Unfortunately the pump I am using has a 2 degree hysteresis and lacks the freezeshield function, But I can live with it.

If you have homeassistant or similar, you can setup an automation to lower the hysteresis and make a freeze shield. Should be possible to make it even to fractions of a degree with the virtual temperature.

**Reply by frogsuk** [reply]
**wyliecoyoteuk**
I purchased one of your boards and have assembled the V1 board.
I was able to connect to the boards AP and put in my wifi details and have just 2.4mhz wifi enabled. I have also now added the jump wires as **visualapproch** mentioned I should do.
The board is now connected physically to my hot tub and when I reconnected the mains supply, it played its happy tune, and the hot tub shows the ip address of 192.168.1.187 and then its temperature. Great, I thought... 
Wrong...

The board is flashing its blue light very fast and I have tried to connect to it via lazyspa.local/index.html and also 192.168.1.187/index.html 
I get this screen
![image](https://user-images.githubusercontent.com/47825078/187085998-8149e524-38e5-4e6a-b3f6-4bd68d600875.png)but nothing. I have also done a Lan scan on my network and it has found the board but has named it
```
Name: ESP-0BC240(2.4GHz)
MAC: 4c:75:25:0b:00:40
IPv4: 192.168.1.187
```
I have also carried out a ping from my iPhone network scanning app and my iMac terminal but they both comes back with 
**timeout
No route to host
Host is down**

So, not sure what is going on and would love it if either of you could shine some light on it for me
**wyliecoyoteuk** Please can you post a picture of your complete V1 board showing the lead plugs as I am not sure if my leads are the correct way around?

**Reply by frogsuk** [reply]
I’ve got it working!!!!
Did a port scan from my iPhone and found port 80 and 81 are open so managed to get into it with the good old fashioned 
192.168.1.187:80😀![image](https://user-images.githubusercontent.com/47825078/187086721-6cd8f384-6876-48c2-863e-d4881c124418.png)![image](https://user-images.githubusercontent.com/47825078/187086728-85eb5dfa-a7f6-4b6d-a098-b01aeb454330.png)

**Reply by frogsuk** [reply]
Spoke to soon😫
No ports open now and not connecting. Thats it for today.
Anaother go tomorrow, but ther is hope 😂
I am going to put another wifi repeater to boost the signal within 2 meters of the hot tub to see if that cures it here. I will report back tomorrow

**Comment by wyliecoyoteuk** [comment]
I would try http://192.168.1.187 as the local DNS may not work.
It is important to use type it using  http NOT https, and some browsers will automatically add the s for you and just say cannot connect.
The blue led flashing is nothing to worry about. here is my board:
![image](https://user-images.githubusercontent.com/79256941/187085899-254a0585-f714-4985-8340-09767605fda6.png)

**Reply by frogsuk** [reply]
Thank you.. Phew my wires are the same but I too have the wifi dropping in and out I think, so tomorrow, as reported above, I am going to add a wifi repeater close to it and fingers crossed

**Reply by wyliecoyoteuk** [reply]
I think that the vodafone hub might be thinking it is an IP phone.

**Comment by wyliecoyoteuk** [comment]
Just got around to retrying this with a V2 board and the latest firmware.
Everything works from the web interface! 
But the physical display and buttons don't. 
(The wifi indicator comes on).
The Bestway app works as well.
I am assuming that there is not enough power getting through.

**Reply by visualapproach** [reply]
I commented in another thread. You're tagged so I think you can find it.

**Reply by visualapproach** [reply]
Hi, I made another PCB with fatter signal tracks. I don't know if it will help, and I haven't tested it myself, but you can find it in the development branch readme

**Comment by wyliecoyoteuk** [comment]
I had to return to this because my original non-wifi egg failed.
I have ordered a v2b board and caps and resistors to try.
However, in the mean time I found this [https://github.com/cdpuk/ha-bestway](https://github.com/cdpuk/ha-bestway) which is an integration for Home assistant that works very well with the Bestway app and API ( much better than the LayZspa mess of an app).
I will be persevering with the hack though, as I prefer to have it local on my networrk.
